packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx

```diff
@@ -0,0 +1,230 @@
+    whiteSpace: 'nowrap',
+  },
+  reactIcon: {
+    paddingRight: "4px",
+  },
+  suggestionRow: {
+    display: 'flex',
+  },
+  agreeText: {
+    color: theme.palette.text.dim3,
+  },
+  agreeTextCollapsedMobile: {
+    [theme.breakpoints.down('xs')]: {
+      display: 'none'
+    },
+  },
+  topicRecommendationsList: {
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    color: theme.palette.text.primary,
+    fontSize: 'small',
+    overflow: 'hidden',
+    justifyContent: 'space-between'
+  },
+  expandIcon: {
+    cursor: 'pointer',
+    minWidth: '70px',
+    color: theme.palette.text.dim3,
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    [theme.breakpoints.down('xs')]: {
+      marginLeft: 'auto',
+      minWidth: '25px',
+    },
+  },
+  hideIcon: {
+    cursor: 'pointer',
+    minWidth: '70px',
+    [theme.breakpoints.down('xs')]: {
+      marginLeft: 'auto',
+      minWidth: '25px',
+    },
+    color: theme.palette.text.dim3,
+    fontFamily: theme.palette.fonts.sansSerifStack,
+  },
+  dialogueRightContainer: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    //paddingRight: 10,
+    //marginRight: 3,
+    marginLeft: 'auto',
+  },
+  dialogueMatchNote: {
+    [theme.breakpoints.down('xs')]: {
+      display: 'none'
+    }
+  },
+  bigScreenExpandNote: {
+    [theme.breakpoints.down('xs')]: {
+      display: 'none'
+    }
+  }
+});
+
+interface DialogueRecommendationRowProps {
+  rowProps: DialogueUserRowProps<boolean>; 
+  classes: ClassesType<typeof styles>; 
+  showSuggestedTopics: boolean;
+}
+
+const DialogueRecommendationRow = ({ rowProps, classes, showSuggestedTopics }: DialogueRecommendationRowProps) => {
+  const { DialogueCheckBox, UsersName, PostsItem2MetaInfo, LWTooltip, ReactionIcon } = Components
+
+  const { targetUser, checkId, userIsChecked, userIsMatched } = rowProps;
+  const { captureEvent } = useTracking(); //it is virtuous to add analytics tracking to new components
+  const currentUser = useCurrentUser();
+  const [isExpanded, setIsExpanded] = useState(false);
+  const toggleExpansion = () => {
+    setIsExpanded(!isExpanded);
+  };
+
+  const { loading, error, data: topicData } = useQuery(gql`
+    query getTopicRecommendations($userId: String!, $targetUserId: String!, $limit: Int!) {
+      GetTwoUserTopicRecommendations(userId: $userId, targetUserId: $targetUserId, limit: $limit) {
+        comment {
+          _id
+          contents {
+            html
+            plaintextMainText
+          }
+        }
+        recommendationReason
+        yourVote
+        theirVote
+      }
+    }
+  `, {
+    variables: { userId: currentUser?._id, targetUserId: targetUser._id, limit:4 },
+  });
+
+  if (!currentUser) return <></>;
+  const preTopicRecommendations: {comment: {_id: string, contents: {html: string, plaintextMainText: string}}, recommendationReason: string, yourVote: string, theirVote: string}[] | undefined = topicData?.GetTwoUserTopicRecommendations; 
```

Comment:
please define an interface for this

Reply:
Actually isn't this just `TopicRecommendation`?  If so you can move it from the server to the client (since you can't import from server files in client files)